nested directory getting lost between steps

hi! I'm unsuccessfully trying to run our CWL pipeline with toil. It looks like somehow intermediate date products get lost. The pipeline runs fine with the CWL reference runner. My logfile is filled with errors like:

```
2017-11-26 11:19:34,177 - toil.leader - WARNING - W/2/jobLmwkno    WorkflowException: [Errno 2] No such file or directory: u'/vault-tina/gijs/prefactor-cwl/runs/run_2017-11-26-11-18-25/results/out_tmpdir2RF79v/calibrated.MS/OBSERVATION/table.lock'
```

The intermediate folder seems to be empty. My guess this is probably related to our data products being nested directory structures and maybe in combination with docker.

To reproduce this issue you could:
* clone our repository https://github.com/EOSC-LOFAR/prefactor-cwl/tree/b5c080d10c659232b46dd9dcb2d9e4c19cc9febc
* run `make toil` in the checkout

Requirements:
 * git-lfs
 * Docker
 * virtualenv

I posted an issue on your repo, but in attempting to use either `make run` or `make toil` I get the same error (running on Ubuntu 16.04):

`lifeisaboutfishtacos@TacoTruck:~/Desktop/LOFAR/prefactor-cwl$ make run`
`cd data && tar Jxvf L570745_uv_first10.MS.tar.xz`
`xz: (stdin): File format not recognized`
`tar: Child returned status 1`
`tar: Error is not recoverable: exiting now`
`Makefile:19: recipe for target 'data/L570745_SB000_uv_first10.MS/' failed`
`make: *** [data/L570745_SB000_uv_first10.MS/] Error 2`

--- a/src/toil/cwl/cwltoil.py
+++ b/src/toil/cwl/cwltoil.py
@@ -455,12 +455,6 @@ class CWLJob(Job):
                                                 cwltool.stdfsaccess.StdFsAccess(outdir),
                                                 recursive=True))
 
-        def make_dir_literal(obj):
-            if "location" in obj and obj["location"].startswith("file:"):
-                obj["location"] = "_:" + str(uuid.uuid4())
-
-        adjustDirObjs(output, make_dir_literal)
-
         adjustFileObjs(output, functools.partial(uploadFile,
                                                  functools.partial(writeGlobalFileWrapper, fileStore),
                                                  index, existing))

